src/nvim/msgpack_rpc/channel.c

```diff
@@ -357,6 +339,30 @@ static void handle_request(Channel *channel, msgpack_object *request)
   }
 }
 
+static MsgpackRpcRequestHandler retrieve_request_handler(msgpack_object *request, Array *args)
```

Comment:
`request` must be validated before we can use it here so freely. Can you add a function comment for the function arguments and what we expect?

Function attribute `FUNC_ATTR_NONNULL_ALL` could be added.

Reply:
Come to think of it, if a method is set after validation it seems quite odd that there is a possibility for `msgpack_rpc_get_handler_for` to return a handler with fn =  `msgpack_rpc_handle_missing_method`. Removing it breaks the test, could you help me with that misunderstanding please?

Reply:
> My goal here would be that this function returns an MsgpackRpcHandler retrieved by msgpack_rpc_get_handler_for and, on error ( invalid arguments or missing method ) returning an noop_handler and setting an error to pass to the event.

Are you describing what the function is already doing here? Otherwise `msgpack_rpc_handle_missing_method` and `msgpack_rpc_handle_invalid_arguments` are the noop functions that encode the error.